components/apps_form/apps_form.tsx

```diff
@@ -47,15 +45,15 @@ type FormResponseData = {
 
 type State = {
     show: boolean;
-    values: {[name: string]: string};
+    values: AppFormValues;
     error: string | null;
     errors: {[name: string]: React.ReactNode};
-    submitting: boolean;
+    submitting: string | null;
```

Comment:
0/5 on this, but I like the Go idiom of not letting strings be null. I think replacing null with the empty string here will have the same behaviour. Otherwise, code looks good.

Reply:
If we don't have `| null`, we lose the type-checking for making sure the value is defined when we want to use it

Reply:
With the way we're using it elsewhere like with `Boolean(this.state.submitting)`, the empty string and null are equivalent though, so null isn't really necessary. Not that it makes much of a difference either way if we use null here or not.

On a related note, I've been favouring undefined over null lately since it's better supported (like this could be abbreviated to `submitting?: string`). I know our ESLint rules prevented that before, but I've been finding it nicer to use.

Reply:
Using undefined definitely makes more sense, especially for the reason it's being used. It's a lot more concise and intention-revealing